src/vm/functions/tuples.rs

```diff
@@ -53,3 +54,18 @@ pub fn tuple_get(args: &[SymbolicExpression], env: &mut Environment, context: &L
         _ => Err(Error::new(ErrType::TypeError("TupleType".to_string(), value.clone())))
     }
 }
+
+pub enum TupleDefinitionType {
+    Implicit(Box<[SymbolicExpression]>),
+    Explicit,
+}
+
+pub fn tuple_definition_type(args: &SymbolicExpression) -> TupleDefinitionType {
```

Comment:
This method should first verify that `args` is a tuple, and if not, should return an `Err(...)`.  As such, the return type of this method should be `Result<TupleDefinitionType>`.

Reply:
I think adding a code comment here that details the expected context for calling this function -- it's just trying to identify whether a symbolic expression is an implicit tuple structure (i.e., a list of pairs), or some other kind of definition. The caller is responsible for any eventual type checks or actual execution.

It's important that it behave this way, or otherwise it wouldn't be able to be used both in the type checker (which doesn't eval the resulting structure, it only type checks it) and the interpreter (which would want to eval the result, and then do type enforcement on the value, not the type signature).

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1007/commits/bc45b8801ed33f128a5e300ab8108e5d1a66879f